docs/overview/06-verification_pallets/00-abstract.md

```diff
@@ -0,0 +1,35 @@
+---
+title: Abstract Verifier
+---
+
+## [`Abstract Verifier`](https://github.com/HorizenLabs/NH-core/tree/main/pallets/verifiers)
+
+The generic verifier abstraction that define the extrisics, events and errors for all verifier. It also
+implements the logic to generate the statements hash. Every verifier should implement the
+`hp_verifiers::Verifier` trait:
+
+- `Proof`: The proof data
+- `Vk`: The verification key data
+- `Pubs`: The public inputs data.
+- `fn verify_proof()` the main verifier logic.
+- `fn hash_context_data()` the unique context that identify the verifier (i.e. `b"fflonk"`)
+- `fn pubs_bytes()` how to get the bytes from public inputs that will be hashed by `keccak256` in the final statement
```

Comment:
It might be worth to add some heads-up to the developer, mentioning that this would be the same hash that he will need to compute on-chain to check that its proof will be part of the attestation. Same goes for `fn vk_bytes()`

Reply:
I add some notes that should close the gap. Let me know.